larray/core/axis.py

```diff
@@ -0,0 +1,2209 @@
+# -*- coding: utf8 -*-
+from __future__ import absolute_import, division, print_function
+
+import re
+import sys
+import warnings
+from itertools import product
+
+import numpy as np
+
+from larray.core.expr import ExprNode
+from larray.core.group import (ABCAxis, ABCAxisReference, ABCLArray, Group, LGroup, PGroup, PGroupMaker,
```

Comment:
those 3 ABC should be in their own "abc" module I think. Having them in group.py seems completely arbitrary.